scipy/stats/unuran/unuran_wrapper.pyx.templ

```diff
@@ -0,0 +1,869 @@
+    max_sqhratio : float, optional
+        Set upper bound for the ratio (area below squeeze) / (area below hat).
+        It must be a number between 0 and 1. Default is 0.99.
+    max_intervals : int, optional
+        Set maximum number of intervals. Default is 100.
+    use_center : bool, optional
+        Use the center as construction point, if given. Default is True.
+    use_mode : bool, optional
+        Use the (exact!) mode as construction point, if given. Default is
+        True.
+    guide_factor : float, optional
+        Set factor for relative size of the guide table for indexed search.
+        It must be greater than or equal to 0. When set to 0, then sequential
+        search is used. Default is 2.
+    seed : int, BitGenerator, Generator, RandomState, SeedSequence, optional
+        Seed for the underlying uniform random number generation.
+
+    Notes
+    -----
+    The transformation of the PDF must be concave in order to
+    construct the hat function. Such a PDF is called T-concave. Currently
+    the following transformations are supported:
+
+    .. math::
+
+        c = 0.: T(x) &= \log(x)\\
+        c = -0.5: T(x) &= \frac{1}{\sqrt{x}} \text{ (Default)}
+
+    Three variants of this method are available:
+
+    * GW: squeezes between construction points.
+    * PS: squeezes proportional to hat function. (Default)
+    * IA: same as variant PS but uses a compositon method with
+      "immediate acceptance" in the region below the squeeze.
+
+    
+    References
+    ----------
+    .. [1] UNU.RAN reference manual, Section 5.3.16,
+           "TDR - Transformed Density Rejection",
+           http://statmath.wu.ac.at/software/unuran/doc/unuran.html#TDR
+    .. [2] Hörmann, Wolfgang. "A rejection technique for sampling from
+           T-concave distributions." ACM Transactions on Mathematical
+           Software (TOMS) 21.2 (1995): 182-193
+
+    Examples
+    --------
+    >>> from scipy.stats import TransformedDensityRejection
+
+    Suppose we have a density:
+
+    .. math::
+
+        f(x) = \begin{cases}
+                1 - x^2,  &  -1 \leq x \leq 1 \\
+                0,        &  \text{otherwise}
+               \end{cases}
+
+    The derivative of this density function is:
+
+    .. math::
+
+        \frac{df(x)}{dx} = \begin{cases}
+                            -2x,  &  -1 \leq x \leq 1 \\
+                            0,    &  \text{otherwise}
+                           \end{cases}
+
+    Notice that the PDF doesn't integrate to 1. As this is a rejection based
+    method, we need not have a normalized PDF. To initialize the generator,
+    we can use:
+
+    >>> urng = np.random.default_rng()
+    >>> pdf = lambda x: 1-x*x
+    >>> dpdf = lambda x: -2*x
+    >>> rng = TransformedDensityRejection(pdf, dpdf, domain=(-1, 1),
+    ...                                   seed=urng)
+
+    Now, we can use the `rvs` method to generate samples from the
+    distribution:
+
+    >>> rvs = rng.rvs(1_000)
```

Comment:
minor point: do we need _ for such small numbers? i have not seen it in other SciPy examples yet. TBD

Reply:
> minor point: do we need _ for such small numbers?

No, we don't. I did that just for my readability. I will change this too in the next commit.

Reply:
I have removed these from the documentation now. I will see if any of these are present in the tests/code and remove them , if present.